src/overlays/actors/ovl_Obj_Tsubo/z_obj_tsubo.c

```diff
@@ -46,39 +73,275 @@ static ColliderCylinderInit D_80BA1B94 = {
+    if (Actor_HasParent(&this->actor, globalCtx)) {
+        ObjTsubo_SetupLiftedUp(this);
+        return;
+    }
+    if ((this->actor.bgCheckFlags & 0x20) && (this->actor.yDistToWater > 15.0f)) {
+        ObjTsubo_WaterBreak(this, globalCtx);
+        Audio_PlaySoundAtPosition(globalCtx, &this->actor.world.pos, 20, NA_SE_EV_POT_BROKEN);
+        ObjTsubo_SpawnCollectible(this, globalCtx);
+        Actor_Kill(&this->actor);
+        return;
+    }
+    if ((this->collider.base.acFlags & AC_HIT) && (this->collider.info.acHitInfo->toucher.dmgFlags & 0x4FC1FFFC)) {
+        ObjTsubo_AirBreak(this, globalCtx);
+        ObjTsubo_SpawnCollectible(this, globalCtx);
+        Audio_PlaySoundAtPosition(globalCtx, &this->actor.world.pos, 20, NA_SE_EV_POT_BROKEN);
+        Actor_Kill(&this->actor);
+        return;
+    }
+    if (this->actor.xzDistToPlayer < 600.0f) {
+        Collider_UpdateCylinder(&this->actor, &this->collider);
+        this->collider.base.acFlags &= ~AC_HIT;
+        CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+        if (this->actor.xzDistToPlayer < 150.0f) {
+            CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+        }
+    }
+    if (this->actor.xzDistToPlayer < 100.0f) {
+        temp_v0 = this->actor.yawTowardsPlayer - PLAYER->actor.world.rot.y;
+        phi_v1 = ABS(temp_v0);
+        if (phi_v1 >= 0x5556) {
+            func_8002F434(&this->actor, globalCtx, 0, 30.0f, 30.0f);
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Tsubo/func_80BA17C4.s")
+void ObjTsubo_SetupLiftedUp(ObjTsubo* this) {
+    this->actionFunc = ObjTsubo_LiftedUp;
+    this->actor.room = -1;
+    func_8002F7DC(&this->actor, NA_SE_PL_PULL_UP_POT);
+    this->actor.flags |= 0x10;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Tsubo/func_80BA180C.s")
+void ObjTsubo_LiftedUp(ObjTsubo* this, GlobalContext* globalCtx) {
+    if (Actor_HasNoParent(&this->actor, globalCtx) != 0) {
+        this->actor.room = globalCtx->roomCtx.curRoom.num;
+        func_80BA188C(this);
+        ObjTsubo_Fall(this);
+        func_8002D7EC(&this->actor);
+        Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 5.0f, 15.0f, 0.0f, 0x85);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Tsubo/func_80BA188C.s")
+void func_80BA188C(ObjTsubo* this) {
+    this->actor.velocity.x = Math_SinS(this->actor.world.rot.y) * this->actor.speedXZ;
+    this->actor.velocity.z = Math_CosS(this->actor.world.rot.y) * this->actor.speedXZ;
+    this->actor.colChkInfo.mass = 0xF0;
+    D_80BA1B50 = (Rand_ZeroOne() - 0.7f) * 2800.0f;
+    D_80BA1B58 = (Rand_ZeroOne() - 0.5f) * 2000.0f;
+    D_80BA1B54 = 0;
+    D_80BA1B5C = 0;
+    this->actionFunc = func_80BA1958;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Tsubo/func_80BA1958.s")
+void func_80BA1958(ObjTsubo* this, GlobalContext* globalCtx) {
```

Comment:
This function handles the pot after Link has thrown or dropped it, if you'd like to give it a name to reflect that.